[2023-08-16T07:51:13.035+0000] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: forex_data_pipeline.saving_rates manual__2023-08-16T07:46:05.693290+00:00 [queued]>
[2023-08-16T07:51:13.046+0000] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: forex_data_pipeline.saving_rates manual__2023-08-16T07:46:05.693290+00:00 [queued]>
[2023-08-16T07:51:13.047+0000] {taskinstance.py:1279} INFO - 
--------------------------------------------------------------------------------
[2023-08-16T07:51:13.047+0000] {taskinstance.py:1280} INFO - Starting attempt 2 of 2
[2023-08-16T07:51:13.048+0000] {taskinstance.py:1281} INFO - 
--------------------------------------------------------------------------------
[2023-08-16T07:51:13.063+0000] {taskinstance.py:1300} INFO - Executing <Task(BashOperator): saving_rates> on 2023-08-16 07:46:05.693290+00:00
[2023-08-16T07:51:13.087+0000] {standard_task_runner.py:55} INFO - Started process 9343 to run task
[2023-08-16T07:51:13.091+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'forex_data_pipeline', 'saving_rates', 'manual__2023-08-16T07:46:05.693290+00:00', '--job-id', '51', '--raw', '--subdir', 'DAGS_FOLDER/forex_data_pipeline.py', '--cfg-path', '/tmp/tmp2rbn2v_b']
[2023-08-16T07:51:13.091+0000] {standard_task_runner.py:83} INFO - Job 51: Subtask saving_rates
[2023-08-16T07:51:13.156+0000] {task_command.py:388} INFO - Running <TaskInstance: forex_data_pipeline.saving_rates manual__2023-08-16T07:46:05.693290+00:00 [running]> on host 2761a966e4c6
[2023-08-16T07:51:13.221+0000] {taskinstance.py:1509} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=admin@localhost.com
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=forex_data_pipeline
AIRFLOW_CTX_TASK_ID=saving_rates
AIRFLOW_CTX_EXECUTION_DATE=2023-08-16T07:46:05.693290+00:00
AIRFLOW_CTX_TRY_NUMBER=2
AIRFLOW_CTX_DAG_RUN_ID=manual__2023-08-16T07:46:05.693290+00:00
[2023-08-16T07:51:13.223+0000] {subprocess.py:63} INFO - Tmp dir root location: 
 /tmp
[2023-08-16T07:51:13.223+0000] {subprocess.py:75} INFO - Running command: ['/bin/bash', '-c', '\n            hdfs dfs -mkdir -p /forex &&             hdfs dfs -put -f $AIRFLOW_HOME/dags/files/forex_rates.json /forex\n            ']
[2023-08-16T07:51:13.234+0000] {subprocess.py:86} INFO - Output:
[2023-08-16T07:51:13.236+0000] {subprocess.py:93} INFO - /bin/bash: line 2: hdfs: command not found
[2023-08-16T07:51:13.236+0000] {subprocess.py:97} INFO - Command exited with return code 127
[2023-08-16T07:51:13.246+0000] {taskinstance.py:1768} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/bash.py", line 197, in execute
    f"Bash command failed. The command returned a non-zero exit code {result.exit_code}."
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 127.
[2023-08-16T07:51:13.250+0000] {taskinstance.py:1323} INFO - Marking task as FAILED. dag_id=forex_data_pipeline, task_id=saving_rates, execution_date=20230816T074605, start_date=20230816T075113, end_date=20230816T075113
[2023-08-16T07:51:13.262+0000] {standard_task_runner.py:105} ERROR - Failed to execute job 51 for task saving_rates (Bash command failed. The command returned a non-zero exit code 127.; 9343)
[2023-08-16T07:51:13.303+0000] {local_task_job.py:208} INFO - Task exited with return code 1
[2023-08-16T07:51:13.318+0000] {taskinstance.py:2578} INFO - 0 downstream tasks scheduled from follow-on schedule check
